server/data/cycles.ts

```diff
@@ -0,0 +1,137 @@
+import { cycles, users } from '../config/mongoCollections';
+import { ObjectId } from 'mongodb';
+import { checkObjectId } from '../helpers/error';
+
+export default {
+
+    /**
+     * @description Get a cycle by id
+     * @param {string} id Cycle id
+     * @returns {Promise<Object>} Cycle object
+     */
+    readById: async (id: string): Promise<object> => {
+        if(!checkObjectId(id)){
+            throw new Error('Invalid id');
+        }
+
+        const cycleCollection = await cycles();
+        const cycle = await cycleCollection.findOne({_id: new ObjectId(id)});
+        if(cycle === null)
+            throw new Error("There is no cycle with that id.");
+
+        cycle['_id'] = cycle['_id'].toString();
+        return cycle;
+    },
+
+    /**
+     * @description Get all cycles for a given user
+     * @param userId User id
+     * @returns List of cycles
+     */
+    readAll: async (userId: string): Promise<object[]> => {
+        if(!checkObjectId(userId)){
+            throw new Error('Invalid id');
+        }
+
+        const cycleCollection: any = await cycles();
+        const userCollection: any = await users();
+        const user: any = await userCollection.findOne({
+            _id: new ObjectId(userId) });
+        if(user === null)
+            throw new Error("There is no user with that id.");
+
+        let userCycles: object[] = [];
+        for(let i = 0; i < user.cycles.length; i++) {
+            const cycle = await cycleCollection.findOne({
+                _id: new ObjectId(user.cycles[i]) });
+            cycle['_id'] = cycle['_id'].toString();
+            userCycles.push(cycle);
+        }
+        return userCycles;
+    },
+
+    /**
+     * @description Create a new cycle
+     * @param userId User id
+     * @returns The newly created cycle object
+     */
+    create: async (userId: string): Promise<object> => {
+        if(!checkObjectId(userId)){
+            throw new Error('Invalid id');
+        }
+
+        let currDate: Date = new Date();
+        let cycle: object = {
+            startDate: currDate,
+            endDate: null,
+            applications: []
+        }
+
+        // Add to cycles document
+        const cycleCollection: any = await cycles();
```

Comment:
would rather not do `:any` and just leave typeless if we do not know it. Stylistic preference, not blocking